### DISCUSSION TITLE
pg_query 2.0 (Postgres parser) released

### DISCUSSION THREAD

**Discussion by Alex-Monahan** [duckdb/duckdb#1980] [Ideas]
It looks like the pganalyze folks have released pg_query 2.0 (and gave a shout out to DuckDB as a user of 1.0!) 

Would there be any benefits to using 2.0 at some point? I saw that they now support parsing multi-statement queries, and they have a query fingerprinting feature that could be used to cache query plans maybe? 

They also support Postgres 13 syntax (up from 10 I believe) with new goodies like: JSON path syntax, stored generated columns, and create procedure syntax. Maybe others?

Just wanted to let you folks know!

https://pganalyze.com/blog/pg-query-2-0-postgres-query-parser

https://www.postgresql.org/about/featurematrix/#sql

**Comment by hannes** [comment]
Well, we heavily patched, refactored, and stripped that excellent parser. At this point its unlikely we will port to any new libpg_query releases. We may add specific syntax extensions if applicable though.

**Reply by Mytherin** [reply]
We are very thankful for the libpg_query folks as the library has been tremendously helpful. However, as Hannes mentioned, we have heavily patched it at this point and extended it with many of our own features, so switching back to the original library is unlikely. We do occasionally still fetch changes from Postgres' yacc file and integrate it with our fork, but that is more on a feature by feature basis.

**Reply by Alex-Monahan** [reply]
Makes sense! Nothing super critical between version 10 and 13 on the SQL syntax side, and DuckDB's enhanced syntax flexibility vs  Postgres is much appreciated!
